kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+
+            core.policy_engine.set_policy(&body).await?;
+
+            Ok(HttpResponse::Ok().finish())
+        }
+        #[cfg(feature = "resource")]
+        "resource" => {
+            if request.method() == Method::GET {
+                // Resource APIs needs to be authorized by the Token and policy
+                let resource_desc =
+                    sub_path
+                        .strip_prefix('/')
+                        .ok_or(Error::IllegalAccessedPath {
+                            path: end_point.clone(),
+                        })?;
+
+                let token = core
+                    .get_attestation_token(&request)
+                    .await
+                    .map_err(|_| Error::TokenNotFound)?;
+
+                let claims = core.token_verifier.verify(token).await?;
+
+                let claim_str = serde_json::to_string(&claims)?;
+                if !core
+                    .policy_engine
+                    .evaluate(resource_desc, &claim_str)
+                    .await?
+                {
+                    return Err(Error::PolicyDeny);
+                };
+
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                let resource = core
+                    .resource_storage
+                    .get_secret_resource(resource_description)
+                    .await?;
+
+                let public_key = core.token_verifier.extract_tee_public_key(claims)?;
+                let jwe = jwe(public_key, resource).map_err(|e| Error::JweError { source: e })?;
+
+                let res = serde_json::to_string(&jwe)?;
+
+                Ok(HttpResponse::Ok()
+                    .content_type("application/json")
+                    .body(res))
+            } else if request.method() == Method::POST {
+                let resource_desc =
+                    sub_path
+                        .strip_prefix('/')
+                        .ok_or(Error::IllegalAccessedPath {
+                            path: end_point.clone(),
+                        })?;
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                core.admin_auth.validate_auth(&request)?;
+                core.resource_storage
+                    .set_secret_resource(resource_description, &body)
+                    .await?;
+
+                Ok(HttpResponse::Ok().content_type("application/json").body(""))
+            } else {
+                Ok(HttpResponse::NotImplemented()
+                    .content_type("application/json")
+                    .body(""))
+            }
+        }
+        plugin_name => {
+            // Plugin calls needs to be authorized by the Token and policy
+            let token = core
+                .get_attestation_token(&request)
+                .await
+                .map_err(|_| Error::TokenNotFound)?;
+
+            let claims = core.token_verifier.verify(token).await?;
```

Comment:
I like that we're decoupling the plugin from the `get_resource` API. It will allow us to implement more complex plugins including ones that take `POST` requests.

I think there is a problem with the logic here. We are passing the request to the handler function in the plugin, allowing it to specify different behavior for `GET` and `POST` requests, but in this function we are verifying the token and the policy for the request regardless of whether it is `GET` or `POST`. Basically this means that this plugin endpoint can't be used to implement requests from an admin.

Maybe we should distinguish between `GET` and `POST` here and do separate checks? Note that there is an open question about what `POST` requests should represent. See my main comment about this.